checks.d/dns_check.py

```diff
@@ -52,21 +52,27 @@ def query_dns(self, instance, timeout):
             resolver.nameservers = [nameserver]
 
         record_type = instance.get('record_type', 'A')
+        expected_results = instance.get('addresses', [])
 
         status = AgentCheck.CRITICAL
         start_time = time.time()
         try:
             self.log.debug('Querying "%s" record for hostname "%s"...' % (record_type, hostname))
             answer = resolver.query(hostname, rdtype=record_type)
+            resolved_value = answer.rrset.items[0].to_text()
             assert(answer.rrset.items[0].to_text())
             end_time = time.time()
+            if len(expected_results) > 0:
+                if resolved_value not in expected_results:
+                    raise Exception('DNS resolution of %s resulted in unexpected address %s.' % (hostname, resolved_value))
```

Comment:
There might be a better way than raising an exception, since this verification can only take place if the DNS resolution was done correctly: put this block after the `else`, before the `if end_time - start_time > 0`, and log an error + service critical. What do you think ?